### PR TITLE
Authorize.net gateway - Issue/204

### PR DESCRIPTION
**Pull request by pippinsplugins**
#204

### CODE REVIEW

File: includes/gateways/class-rcp-payment-gateway-authorizenet.php [hunk 1]
```diff
@@ -125,6 +126,11 @@ public function process_signup() {
 			$subscription->billToLastName = $lname;
 			$subscription->billToZip = sanitize_text_field( $_POST['rcp_card_zip'] );
 
+			// Delay start date for free trials.
```
Comment (pippinsplugins) [inline includes/gateways/class-rcp-payment-gateway-authorizenet.php]: Authorize.net has a dedicated trial period / amount field. Could we use that instead?

$subscription->totalOccurrences = $this->auto_renew ? 9999 : 1;
$subscription->trialOccurrences = $this->auto_renew ? 1 : 0;
$subscription->amount = $this->amount;
$subscription->trialAmount = $this->initial_amount;

Comment (nosegraze) [inline includes/gateways/class-rcp-payment-gateway-authorizenet.php]: I'm not sure you can set a separate duration for that though?

Comment (pippinsplugins) [inline includes/gateways/class-rcp-payment-gateway-authorizenet.php]: You appear to be correct! Good catch.

File: includes/gateways/authorizenet/functions.php [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+ */
+function rcp_has_authnet_api_access() {
+
+	global $rcp_options;
+
+	$ret = false;
+
+	if ( rcp_is_sandbox() ) {
+		$api_login_id   = $rcp_options['authorize_test_api_login'];
+		$transation_key = $rcp_options['authorize_test_txn_key'];
```
Comment (mindctrl) [inline includes/gateways/authorizenet/functions.php]: @pippinsplugins @nosegraze small thing, but `transation_key` is misspelled in all these instances.

Comment (nosegraze) [inline includes/gateways/authorizenet/functions.php]: Lol thanks. Fixing now.

File: includes/member-functions.php [hunk 3]
```diff
@@ -1308,6 +1327,10 @@ function rcp_member_can_update_billing_card( $user_id = 0 ) {
 
 		$ret = true;
 
+	} elseif ( rcp_is_authnet_subscriber( $user_id ) && ! empty( $rcp_options['authorize_api_login'] ) && ! empty( $rcp_options['authorize_txn_key'] ) ) {
+
```
Comment (mindctrl) [inline includes/member-functions.php]: These rcp_options calls can be changed to use `rcp_has_authnet_api_access()`.

Comment (nosegraze) [inline includes/member-functions.php]: Good catch!

**Pull request merged**